### ISSUE TITLE
Error sometimes when creating project

### ISSUE THREAD

**Issue by myieye** [sillsdev/languageforge-lexbox#173]
Sometimes (I think about 40% of the time) I get an error when I get redirected to the project page after creating a new project:
![image](https://github.com/sillsdev/languageforge-lexbox/assets/12587509/db950c94-346a-4861-abba-398bc3fb2dd3)

I think it's just a race condition: the hg repository isn't necessarily ready when the project page asks for the repo change-sets.

Here's the trace: https://ui.honeycomb.io/sil-language-forge/environments/test/trace/5f7iwR6T2AT?fields[]=c_name&fields[]=c_service.name&span=20635f6224b0e3d9

And part of the HG web error: `mercurial.templateutil.TemplateNotFound: "notfound" not in template map`

**Comment by hahn-kev** [comment]
I wrote an API test to try and reproduce this issue locally but was unable to reproduce it. I'm going to try the test against staging.

**Comment by hahn-kev** [comment]
After taking a look at #214 I dug into the mercurial source code. It looks like [this is the function](https://github.com/indygreg/hg/blob/e1b661b1b1edd2cefecd4d305ddee923d0967c53/mercurial/hgweb/hgwebdir_mod.py#L277) that controls the refresh. It looks like there's no way to force it, and no other way to notify it of a repo being added. I think our best bet here will just to be resilient of errors for newly created repos. Maybe we try to get the change log, but if the project is not 1 minute old then we ignore any errors we get from hgweb and just return an empty list.

**Comment by myieye** [comment]
As disussed with @hahn-kev:
We decided that we're just going to try to avoid accessing Hg repos shortly after they've been created.

There are 2 reasons for this:
1) HgWeb has a `refreshinterval` (20s by default). Unless we reduce the interval to 0, there's always the chance that a repo won't be immediately available after being created.
2) Repos appear to be unstable for a short time after they've been created. We observed initially getting 200's, but then shortly after get 404's for a while. It's as if Hg does something with the new repo making it temporarily unavailable.
